src/views/messages/container.jsx

```diff
@@ -116,10 +116,20 @@ var Messages = React.createClass({
             )
         );
     },
-    filterMessages: function (messages, typesAllowed) {
+    filterMessages: function (messages, typesAllowed, unreadCount) {
         var filteredMessages = [];
         for (var i in messages) {
-            if (typesAllowed.indexOf(messages[i].type) > -1) {
+            // check to see if the position of the message in the list is earlier
+            // than the unread count. If it is, then the message is totally unread.
+            messages[i].unread = false;
+            if (i < unreadCount) messages[i].unread = true;
+
+            if (typesAllowed.length > 0) {
+                if (typesAllowed.indexOf(messages[i].type) > -1) {
+                    filteredMessages.push(messages[i]);
+                }
+            } else {
+                // if empty, then we're looking at all messages, so just like add the message
```

Comment:
I wanted to figure out if there was a way not to iterate over the whole list if typesAllowed is empty. Would it work to copy messages to filteredMessages, and then just set  `filteredMessages[i].unread = true` for the first unreadCount ones. Then the mod-unread class could be set based on whether unread is defined and true: `var className = (typeof message.unread === 'boolean' && message.unread) ? ...`